planner/core/task.go

```diff
@@ -1289,6 +1300,82 @@ func BuildFinalModeAggregation(
 	return
 }
 
+// convertAvgForMPP converts avg() to if(count(arg)=0, null, sum(arg)/count(arg)), in detail:
+// 1.rewrite avg() in the final aggregation to count() and sum(), and reconstruct its schema.
+// 2.replace avg() with if(count(arg)=0, null, sum(arg)/count(arg)) and reuse the original schema of the final aggregation.
+func (p *basePhysicalAgg) convertAvgForMPP(prop *property.PhysicalProperty) *PhysicalProjection {
```

Comment:
Why do we need a prop? IMO, the prop in projection plan is not a must.

Reply:
it seems not a must, but in case of using `p.childrenReqProps` for the generating physical plan. This case may occur in`postOptimize()`.

Reply:
Can you figure out which post optimize needs child req props? That makes no sense.

Reply:
`InjectExtraProjection` in the postOptimize()

Reply:
Seems that Child Prop is useless. Why not just pass "p"'s child prop instead of constructing a new one?

Reply:
changed.